times_reader/transforms.py

```diff
@@ -353,50 +351,52 @@ def remove_comment_rows(table: datatypes.EmbeddedXlTable) -> datatypes.EmbeddedX
         return table
 
     df = table.dataframe.copy()
-    comment_rows = list(
-        locate(
-            df.iloc[:, 0],
-            lambda cell: isinstance(cell, str)
-            and (cell.startswith("*") or cell.startswith("\\I:")),
-        )
-    )
-    df.drop(index=comment_rows, inplace=True)
-    df.reset_index(drop=True, inplace=True)
+    colname = df.columns[0]
+
+    tag = table.tag.split(":")[0]
+
+    if tag in config.row_comment_chars:
+        chars_by_colname = config.row_comment_chars[tag]
+    else:
+        return table
 
-    # TODO: the deletion of this third column is a bug. Removing it causes the
-    # program to fail parse all rows. We need to fix the parsing so it can read
-    # all rows and remove this code block.
-    if df.shape[1] > 1:
-        comment_rows = list(
-            locate(
-                df.iloc[:, 1],
-                lambda cell: isinstance(cell, str) and cell.startswith("*"),
+    for colname in df.columns:
+        if colname in chars_by_colname.keys():
+            comment_rows = list(
```

Comment:
The function docstring says rows starting with a comment char in their *first* column are deleted, but it looks like this code might be deleting rows that have a cell starting with a comment char in *any* column?